### PR TITLE
Add API documentation to README

### LINKED ISSUE #23: Add API documentation to README
Hello :wave:

The goal of the task is to add API documentation to the README. Current API is a bit obscure and can be quite complicated to understand at first look.

Example of API documentation can be found in another project like [RC](https://github.com/NodeSecure/rc#api).

### PR DESCRIPTION
**Pull request by fabnguess**
This PR aims to resolve the issue #23 

close #23

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -32,6 +32,13 @@ const domainExpirationDate = await whois(myDomain);
 const mxRecords = await resolveMxRecords(myDomain);
 ```
 
+## API
+
+### whois(): string
```
Comment (PierreDemailly) [inline README.md:37]: Add a description and documentation for parameters.
```suggestion
### whois(domain: string, server = "whois.iana.org"): string
```

File: README.md [hunk 2]
```diff
@@ -32,6 +32,13 @@ const domainExpirationDate = await whois(myDomain);
...
+
+### resolveMxRecords(): (Array<string> | { error: string, context: Error })
```
Comment (PierreDemailly) [inline README.md:39]:
```suggestion
### resolveMxRecords(domain: string): (Array<string> | { error: string, context: Error })
```

**Review by PierreDemailly — APPROVED** [review]
Lgtm
Note: don't need to draft when the PR is not WIP @fabnguess ;)

**Comment by fabnguess** [comment]
> Lgtm Note: don't need to draft when the PR is not WIP @fabnguess ;)

**Pull request closed without merging**